File: ans/loop.c
/*
 * C言語入門4
 * loop.c
 * for文とwhile文の使い方
*/

#include <stdio.h>

int main(void)
{
    int i;

    // for文
    // 0 - 9 までの数字を表示
    printf("for文でのループ\n");
    for (i = 0; i < 10; i++) {
        printf("%d ", i);
    }
    printf("\n");

    // while文
    // 0 - 9 までの数字を表示
    printf("while文でのループ\n");
    i = 0;
    while (i < 10) {
        printf("%d ", i);
        i++;
    }
    printf("\n");

    // do-while文
    // 10を表示
    printf("do-while文でのループ\n");
    do {
        printf("%d ", i);
        i++;
    } while (i < 10);
    printf("\n");

    // break文
    // 0 - 4 までの数字を表示
    printf("break文の使い方\n");
    i = 0;
    for (;;) {
        if (i == 5) {
            break;
        }
        printf("%d ", i);
        i++;
    }
    printf("\n");

    // continue文
    // 0 - 9 までの数字を表示(5を除く)
    printf("continue文の使い方\n");
    for (i = 0; i < 10; i++) {
        if (i == 5) {
            continue;
        }
        printf("%d ", i);
    }
    printf("\n");

    // 多重ループ
    // 0 - 99 までの数字を表示
    printf("多重ループ\n");
    for (i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            if (j == 5) {
                continue;
            }

            if (i == 5) {
                break;
            }
            printf("%d ", i * 10 + j);
        }

        if(i == 2) continue;
        if(i == 9) break;
    }
    printf("\n");

    return 0;
}

File: python/loop.py
# ループ処理

print('for loop')
for i in range(10):
    print(i)

i = 0
print('while loop')
while i < 10:
    print(i)
    i += 1

print('do-while loopはない')
print('break')
for _ in iter(int, 1):
    if _ == 5:
        break
    print(_)

print('continue')
i = 0
while True:
    if i == 5:
        i += 1
        continue

    print(i)
    i += 1
    if i == 10:
        break

print('多重ループ')
for i in range(10):
    for j in range(10):
        if j == 5:
            continue
        
        if i == 5:
            break
        print(i, j)
    
    if i == 2:
        continue
    if i == 9:
        break
